src/abstractprobprog.jl

```diff
@@ -80,3 +81,29 @@ end
 function Base.rand(model::AbstractProbabilisticProgram)
     return rand(Random.default_rng(), NamedTuple, model)
 end
+
+"""
+    predict(
+        [rng::AbstractRNG=Random.default_rng(),]
+        [T=NamedTuple,]
+        model::AbstractProbabilisticProgram,
+        params,
+    ) -> T
+
+Draw a sample from the joint distribution specified by `model` conditioned on the values in
```

Comment:
"condition" isn't quite right. If the model has a probabilistic graph `x -> y -> z`, and `params` contains `y`, this will sample `x ~ p(x)` and `z ~ p(z | y)`, when conditioning would give `x ~ p(x | y)` and `z ~ p(z | y)`

Reply:
I guess we actually want a `do` operator here instead of `condition`.

Reply:
This is a good point.

> I guess we actually want a do operator here instead of condition.

But doesn't `do` cut the graph too, and so it would also remove the `x` completely?

Reply:
> But doesn't `do` cut the graph too, and so it would also remove the `x` completely?

As I understand it, `do(y=something)` only deletes the edges going into `y` and sets `y' to the specified value. The rest of the graph is unchanged; in particular, no nodes are deleted. So `x` would be kept.

Reply:
Ah yes that makes sense.

Okay so we basically want `do`to be like `condition` but without including the accumulation of it's (log) prob?